### PR TITLE
#123/Create controller to update the user avatar

### PR DESCRIPTION
**Pull request by gbudau**
PUT [/api/v1/users/avatar](http://localhost:3000/api#/users/UserController_uploadAvatar) -> update the avatar of the currently authenticated user

image

Subtask of #123

### CODE REVIEW

File: transcendence-app/src/user/user.service.ts [hunk 1]
```diff
@@ -53,4 +59,78 @@ export class UserService {
...
+      disposition: `inline; filename="${fileDto.filename}"`,
+      length: fileDto.size,
+    });
+  }
+
+  async validateAvatarType(path: string): Promise<boolean | undefined> {
+    /**
+     * Import 'file-type' ES-Module in CommonJS Node.js module
+     */
+    const { fileTypeFromFile } = await (eval('import("file-type")') as Promise<
```
Comment (Damrod) [inline transcendence-app/src/user/user.service.ts:125]: >Warning: Executing JavaScript from a string is an enormous security risk. It is far too easy for a bad actor to run arbitrary code when you use eval(). See Never use eval()! , below.

[Here.](https://developer.mozilla.org/en-US/docs/web/javascript/reference/global_objects/eval#never_use_eval!)

**Comment by Damrod** [comment]
Why do we need to use eval there?

File: transcendence-app/src/user/user.service.ts (hunk 1 above)
Comment (gbudau) [reply transcendence-app/src/user/user.service.ts:142]: The `file-type` library uses ES modules, and the NestJS compiles into CommonJS modules, which converts the `import` syntax to `require`.

We use `eval` to import the library dynamically at runtime.

Comment (gbudau) [reply transcendence-app/src/user/user.service.ts:142]: Maybe we could use something else :thinking:
I will do some research.

**Review by Damrod — APPROVED** [review]
Cuando estés conforme, mergeas.

File: transcendence-app/src/user/user.service.ts (hunk 1 above)
Comment (gbudau) [reply transcendence-app/src/user/user.service.ts]: Fixed, now we use `Function()` instead of `eval()`. Thank you :rocket:

**Comment by gbudau** [comment]
Thank you! :tada:

**Review by fizcamposancos — COMMENTED** [review]
LGTM!

**Pull request merged**